src/tools/networkTools.cpp

```diff
@@ -212,39 +212,38 @@ std::map<std::string, std::string> getNetworkInterfaces() {
 }
 
 
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps() {
+  IpAddress bestIp = IpAddress{"127.0.0.1","::1"};
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
 
 #ifndef _WIN32
   const char* const prioritizedNames[] =
       { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
   for(auto name: prioritizedNames) {
     auto it=interfaces.find(name);
-    if(it != interfaces.end() && !(ipv6 && (*it).second.addr6.empty())) {
-      bestPublicIp = (*it).second;
-      break;
+    if (it != interfaces.end()) {
+      if (it->second.addr.find(':') == std::string::npos) bestIp.addr = it->second.addr;
+      else bestIp.addr6 = it->second.addr6;
     }
   }
 #endif
 
   const char* const prefixes[] = { "192.168", "172.16.", "10.0" };
   for(auto prefix : prefixes){
     for(auto& itr : interfaces) {
-      std::string interfaceIp(itr.second.addr);
-      if (interfaceIp.find(prefix) == 0 && !(ipv6 && itr.second.addr6.empty())) {
-        bestPublicIp = itr.second;
-        break;
-      }
+      if (itr.second.addr.find(':') == std::string::npos && itr.second.addr.find(prefix) == 0)
```

Comment:
Why would `itr.second.addr` (which is supposed to be an IPv4 address) contain a `:`?